subsys/modem/modem_cmux.c

```diff
@@ -842,7 +842,7 @@ static void modem_cmux_receive_handler(struct k_work *item)
 {
 	struct k_work_delayable *dwork = k_work_delayable_from_work(item);
 	struct modem_cmux *cmux = CONTAINER_OF(dwork, struct modem_cmux, receive_work);
-	uint8_t buf[16];
+	uint8_t buf[32];
```

Comment:
Should there be a define for this value? Might make it more clear why this size was chosen.

Reply:
Yeah, could be. I wanted to increase the size of this buffer to further reduce the number of calls to `modem_pipe_receive()` (which is what the ISR UART improvement in this PR is about).
I took 32 as I saw that's what `modem_chat` uses for its `work_buf`, and to keep it still relatively small as here the buffer lies on the stack.

The buffer size could even be made configurable in my opinion. What do you think? @bjarki-trackunit

Reply:
This buffer is a bit ugly, the reason I added it to the stack here was to avoid adding yet another buffer one would have to allocate and pass when initializing the cmux instance. The best option is probably to just add the buffer, naming it cache_buffer or something, but Kconfig is definately the simplest option :)

Reply:
I was rather thinking of adding the buffer to the `modem_cmux` struct. Any objection?

Reply:
oh sure, that is way better, that and the Kconfig is perfect :)